corehq/apps/sms/views.py

```diff
@@ -1915,6 +1918,30 @@ def dispatch(self, request, *args, **kwargs):
         return super(SMSSettingsView, self).dispatch(request, *args, **kwargs)
 
 
+@method_decorator(toggles.ICDS_CUSTOM_SMS_REPORT.required_decorator(), name='dispatch')
+class SMSUsageReport(BaseMessagingSectionView):
+    template_name = 'sms/usage_report.html'
+    urlname = 'sms_usage_report'
+    page_title = _('SMS Usage Report')
+
+    @use_daterangepicker
+    def dispatch(self, *args, **kwargs):
+        return super(SMSUsageReport, self).dispatch(*args, **kwargs)
+
+    def post(self, request, *args, **kwargs):
+        try:
+            start_date, end_date = request.POST.get('date_range').split(' to ')
+            from corehq.messaging.scheduling.forms import validate_date
+            validate_date(start_date)
+            validate_date(end_date)
+            send_custom_sms_report.delay(start_date, end_date)
+            messages.success(self.request, _("Report will we soon emailed to the configured users"))
```

Comment:
Are we sending the email to all user? I think it should just go to the user who is submitting this request.

Reply:
Done